beluga/test/beluga/sensor/test_ndt_model.cpp

```diff
@@ -130,7 +132,85 @@ TEST(NDTSensorModelTests, ToCellsNotEnoughPointsInCell) {
   ASSERT_EQ(cells.size(), 0UL);
 }
 
-TEST(NDTSensorModelTests, SensorModel) {
+TEST(NDTSensorModel3DTests, ToCellsNotEnoughPointsInCell) {
+  const double map_res = 0.5;
+  const std::vector map_data{
+      Eigen::Vector3d{0.1, 0.2, 0.0},
+      Eigen::Vector3d{0.112, 0.22, 0.0},
+      Eigen::Vector3d{0.15, 0.23, 0.0},
+  };
+  const auto cells = detail::to_cells<3UL>(map_data, map_res);
+  ASSERT_EQ(cells.size(), 0UL);
+}
+
+TEST(NDTSensorModel3DTests, FitPoints) {
+  {
+    const std::vector meas{
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+    };
+    auto cell = detail::fit_points<3>(meas);
```

Comment:
@serraramiro1 nit: doesn't template parameter deduction work here? Same for `to_cells()`.